### PR TITLE
Feature/csl939 generate testnet addrs

### PR DESCRIPTION
**Pull request by volhovM**
This mostly adds blacklisting to keygen. Also `Testnet` was simplpfied a little bit. Also i've added little bit of cosmetics here and there to crypto modules.

### CODE REVIEW

File: src/keygen/Avvm.hs [hunk 1]
```diff
@@ -75,24 +108,36 @@ genGenesis avvm genCerts holder = GenesisData
...
-                \THIS IS SERIOUS."
-        addressHash . fst <$> keyGen
+getHolderId (Just fileName) = do
+    mSk <- view usPrimKey <$> readUserSecret fileName
+    let sk = fromMaybe (error "No secret key is found in file") mSk
+    pure $ addressHash $ toPublic sk
+getHolderId Nothing = do
+    skPath <- ("redeemingHolderKey" <>) . show <$> randomRIO (0,100000::Int)
+    (pk,sk) <- keyGen
+    putText $ "USING RANDOM STAKEHOLDER ID, WRITING KEY TO " <> fromString skPath
```
Comment (georgeee) [inline src/keygen/Avvm.hs]: Love you

File: src/Pos/Block/Worker.hs [hunk 2]
```diff
@@ -110,7 +109,7 @@ blkOnNewSlotImpl (slotId@SlotId {..}) sendActions = do
                                 in siEpoch >= w0 && siEpoch <= w1) proxyCerts
             validCert = find (\pSk -> addressHash (pskIssuerPk pSk) == leader)
                              validCerts
-        logNoticeS "THIS IS A SECRET MESSAGE SHOULDN'T GET TO PUBLIC LOGGER"
+        logNoticeS "This is a test debug message which shouldn't be sent to the logging server."
```
Comment (gromakovsky) [inline src/Pos/Block/Worker.hs]: 👍

File: src/keygen/Main.hs [hunk 3]
```diff
@@ -12,19 +12,20 @@ import           System.FilePath      (takeDirectory)
...
 import           KeygenOptions        (AvvmStakeOptions (..), KeygenOptions (..),
                                        TestStakeOptions (..), optsInfo)
 import           Testnet              (genTestnetStakes, generateKeyfile)
 
 replace :: FilePath -> FilePath -> FilePath -> FilePath
 replace a b = toString . (T.replace `on` toText) a b . toText
 
-getTestnetGenesis :: Coin -> TestStakeOptions -> IO GenesisData
-getTestnetGenesis avvmStake tso@TestStakeOptions{..} = do
+getTestnetGenesis :: TestStakeOptions -> IO GenesisData
```
Comment (flyingleafe) [inline src/keygen/Main.hs]: You know that not considering avvm stake while taking N% of stake for richmen is formally incorrect =\

File: src/keygen/KeygenOptions.hs [hunk 4]
```diff
@@ -85,6 +86,11 @@ avvmStakeParser = do
         metavar "FILE" <>
         help    "A keyfile from which to read public key of stakeholder \
                 \to which AVVM stakes are delegated."
+    asoBlacklisted <- optional $ strOption $
+        long    "blacklisted" <>
+        metavar "FILE" <>
+        help    "Path to the file containing blacklisted addresse \
```
Comment (flyingleafe) [inline src/keygen/KeygenOptions.hs]: `addresseS`

File: src/keygen/Main.hs [hunk 5]
```diff
@@ -43,16 +44,21 @@ getTestnetGenesis avvmStake tso@TestStakeOptions{..} = do
             , gdDistribution = distr
             , gdVssCertificates = genesisVssCerts
             }
+
+    putText $ "Total testnet genesis stake: " <> show distr
     return genData
 
 getAvvmGenesis :: AvvmStakeOptions -> IO GenesisData
-getAvvmGenesis AvvmStakeOptions{..} = do
+getAvvmGenesis AvvmStakeOptions {..} = do
```
Comment (flyingleafe) [inline src/keygen/Main.hs]: omg, well, why didn't you change it like this everywhere then?

Comment (volhovM) [inline src/keygen/Main.hs]: `hindent` 🤷‍

File: src/keygen/Main.hs (hunk 3 above)
Comment (volhovM) [inline src/keygen/Main.hs]: It's not a viable claim. I've changed the model to specify only share of testnet stake which i find more obvious. Both work (well in fact i didn't figure out how to use previous implementation, 2 bad for me).

File: core/Pos/Crypto/HD.hs [hunk 6]
```diff
@@ -73,11 +75,9 @@ deriveHDPublicKey (PublicKey xpub) childIndex
 
 -- | Derive secret key from secret key.
 -- If @childIndex <= maxHardened@ key will be deriving hardened way, otherwise non-hardened.
-deriveHDSecretKey :: ByteArrayAccess passPhrase
```
Comment (jagajaga) [inline core/Pos/Crypto/HD.hs]: This was better.

Comment (volhovM) [inline core/Pos/Crypto/HD.hs]: `hindent` ~ style guide 🤷‍

File: src/keygen/Avvm.hs [hunk 7]
```diff
@@ -75,24 +108,36 @@ genGenesis avvm genCerts holder = GenesisData
...
-        putText "IF YOU INTEND TO GENERATE GENESIS.BIN FOR PRODUCTION, \
-                \STOP RIGHT HERE AND USE `--fileholder <path to secret>` OPTION. \
-                \THIS IS SERIOUS."
-        addressHash . fst <$> keyGen
+getHolderId (Just fileName) = do
+    mSk <- view usPrimKey <$> readUserSecret fileName
+    let sk = fromMaybe (error "No secret key is found in file") mSk
+    pure $ addressHash $ toPublic sk
+getHolderId Nothing = do
+    skPath <- ("redeemingHolderKey" <>) . show <$> randomRIO (0,100000::Int)
```
Comment (ChShersh) [inline src/keygen/Avvm.hs]: I think there should be nice way to convert `Int` to `Text` w/o `show` and further conversion from `String` to `Text`

Comment (Martoon-00) [inline src/keygen/Avvm.hs]: `sformat int`? Too many letters :/

**Pull request merged**